Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -105,29 +105,78 @@ public IEnumerable<ParticipantStatusExtendedViewModel> GetParticipantStatusHisto
-                                pt => pt.TeamID,
-                                t => t.ID,
-                                (pt, t) => new TeamExtendedViewModel
-                                {
-                                    ID = t.ID,
-                                    Activity = context.Activity.FirstOrDefault(a => a.ID == t.ActivityID),
-                                    Name = t.Name,
-                                    Status = context.TeamStatus
-                                                .FirstOrDefault(ts => ts.ID == t.StatusID)
-                                                .Description,
-                                    Logo = t.Logo,
-                                    TeamRecord = context.SeriesTeam
-                                        .Include(st => st.Team)
-                                        .Where(st => st.TeamID == t.ID)
-                                        .Select(st => (TeamRecordViewModel)st)
-                                
-                                });
+        var teams = context.ParticipantTeam
+                 .Where(pt => pt.ParticipantUsername == username && pt.RoleTypeID != 0 && pt.RoleTypeID != 2)
+                     .Join(context.Team.Where(t => t.StatusID != 0),
+                         pt => pt.TeamID,
+                         t => t.ID,
+                         (pt, t) => new TeamExtendedViewModel
+                         {
+                             ID = t.ID,
+                             Activity = context.Activity.FirstOrDefault(a => a.ID == t.ActivityID),
+                             Name = t.Name,
+                             Status = context.TeamStatus
+                                         .FirstOrDefault(ts => ts.ID == t.StatusID)
+                                         .Description,
+                             Logo = t.Logo,
+                             TeamRecord = context.SeriesTeam
+                                 .Include(st => st.Team)
+                                 .Where(st => st.TeamID == t.ID)
+                                 .Select(st => (TeamRecordViewModel)st)
+
+                         });
         return teams;
     }
 
+    public IEnumerable<MatchExtendedViewModel> GetParticipantMatches(string username)
+    {
+        var matches = context.ParticipantTeam
+            .Where(pt => pt.ParticipantUsername == username && pt.RoleTypeID != 0 && pt.RoleTypeID != 2)
+                .Join(context.MatchTeam.Where(mt => mt.StatusID != 0 && mt.Match.StatusID != 0)
+                                        .Include(mt => mt.Match)
+                                            .ThenInclude(m => m.Series)
+                                                .ThenInclude(s => s.Activity)
+                                        .Include(mt => mt.Match)
+                                            .ThenInclude(m => m.MatchTeam)
+                                                .ThenInclude(mt => mt.Match)
```

Comment:
Is this not a circular include? Aren't you including `MatchTeam.Match.MatchTeam.Match`?

Reply:
Not EXACTLY, the reason I need any of these includes is because of the implicit operators on the viewmodels. Unless specified, they do not actually hold any of their perceived connections.

The logic flow unfortunately right now is as follows:

MatchTeam(1) is connected via a participant's current team list,
I then get the Match from that MatchTeam(1)
I then need ALL MatchTeam(1->x) connected to the Match
The implicit operator is used on the MatchTeam(1->x) which actually loses its connection to Match and thus loses its properties. 

So yes, it's "circular" but not really?